R/extractData2.R

```diff
@@ -92,9 +92,19 @@ extractData2.GADSdat <- function(GADSdat, convertMiss = TRUE, labels2character =
 }
 
 #'@export
-extractData2.trend_GADSdat <- function(GADSdat, convertMiss = TRUE, labels2character = namesGADS(GADSdat), labels2factor = NULL,
+extractData2.trend_GADSdat <- function(GADSdat, convertMiss = TRUE, labels2character = NULL, labels2factor = NULL,
                                        labels2ordered = NULL, dropPartialLabels = TRUE) {
-  stop("extractData2 has not been implemented for 'trend_GADSdat' objects.")
+  check_trend_GADSdat(GADSdat)
+
+  dat_list <- lapply(names(GADSdat$datList), function(nam) {
+    gads <- extractGADSdat(all_GADSdat = GADSdat, name = nam)
+    dat <- extractData2(gads, convertMiss = convertMiss, labels2character = labels2character, labels2factor = labels2factor,
+                       labels2ordered = labels2ordered, dropPartialLabels = dropPartialLabels)
```

Comment:
Nit: For the sake of readability, I would use a new line for every argument (also holds for the function definition).

``` r
dat <- extractData2(gads, 
                    convertMiss = convertMiss, 
                    labels2character = labels2character, 
                    labels2factor = labels2factor,
                    labels2ordered = labels2ordered, 
                    dropPartialLabels = dropPartialLabels)                     
```

Reply:
The rest: LGTM!

Reply:
Fixed! (although I am not sure whether you meant something else with function definition?)

Reply:
Sorry, that was imprecise: I meant the function assignment (but you already addressed it with your commit):

``` r
fun_name <- function(arg1 = NULL,
                     arg2 = NULL,
                     arg3 = NULL) { ... }
```